Warn when executing Gradle inside an undefined build (#10874)
gradle/gradle: subprojects/docs/src/docs/userguide/upgrading_version_5.adoc

Only the `runtime` configuration now contains more information which corresponds to the _runtimeElements_ variant of a Java library.
In general, users are advised to migrate from `ivy.xml` to the new Gradle Module Metadata format.

> ==== Executing Gradle inside an undefined build has been deprecated

Reviewer: ```suggestion
==== Executing Gradle without a settings file has been deprecated
```

> A Gradle build is defined by the `settings.gradle[.kts]` file in the current or any parent directories.

Reviewer: ```suggestion
A Gradle build is defined by a `settings.gradle[.kts]` file in the current or a parent directory. Without a settings file, a Gradle build is undefined and will emit a deprecation warning. 

In Gradle 7.0, Gradle will only allow you to invoke the `init` task or diagnostic command line flags, such as `--version`, with undefined builds.
```

> Failing to meet this requirement, Gradle is facing an undefined build.

Reviewer: ```suggestion
```

> Executing Gradle in this scenario is only valid when invoking the `init` task, requesting help (either via `--help` or `help` task only for `init` task), or any diagnostic command line flags such as `--version`.

Reviewer: ```suggestion
```
